com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -81,34 +80,20 @@ protected StepPhase executeAsyncStep(ExecutionWrapper execution) throws Exceptio
         }
     }
 
-    private Map<String, String> getServicesGuids(List<String> services, ExecutionWrapper execution) {
-        String spaceId = StepsUtil.getSpaceId(execution.getContext());
+    private Map<String, CloudServiceExtended> getServicesData(List<String> services, ExecutionWrapper execution) {
         CloudControllerClient client = execution.getControllerClient();
 
-        return services.stream()
-            .map(name -> serviceGetter.getServiceInstance(client, name, spaceId))
-            .filter(instance -> instance != null && !instance.isEmpty())
-            .collect(Collectors.toMap(this::getNameFromInstance, this::getGuidFromInstance));
+        return services.parallelStream()
+            .map(name -> client.getService(name))

```

Comment:
Use a method reference: `client:;getService` and rename the `services` variable to `serviceNames` to fix the ambiguity created by the first change.